crates/auth/src/providers/impls/near_wallet.rs

```
@@ -0,0 +1,773 @@
+
+    fn verify_request(&self, request: &Request<Body>) -> eyre::Result<AuthRequestVerifier> {
+        let headers = request.headers();
+
+        // Extract all needed data from the request
+        let account_id = headers
+            .get("x-near-account-id")
+            .ok_or_else(|| eyre::eyre!("Missing NEAR account ID"))?
+            .to_str()
+            .map_err(|_| eyre::eyre!("Invalid NEAR account ID"))?
+            .to_string();
+
+        let public_key = headers
+            .get("x-near-public-key")
+            .ok_or_else(|| eyre::eyre!("Missing NEAR public key"))?
+            .to_str()
+            .map_err(|_| eyre::eyre!("Invalid NEAR public key"))?
+            .to_string();
+
+        let signature = headers
+            .get("x-near-signature")
+            .ok_or_else(|| eyre::eyre!("Missing NEAR signature"))?
+            .to_str()
+            .map_err(|_| eyre::eyre!("Invalid NEAR signature"))?
+            .to_string();
+
+        let message = headers
+            .get("x-near-message")
+            .ok_or_else(|| eyre::eyre!("Missing NEAR message"))?
+            .to_str()
+            .map_err(|_| eyre::eyre!("Invalid NEAR message"))?
+            .to_string();
+
+        // Create auth data
+        let auth_data = NearWalletAuthData {
+            account_id,
+            public_key,
+            message,
+            signature,
+            recipient: String::new(),
+        };
+
+        // Create verifier
+        let provider = Arc::new(self.clone());
+        let verifier = NearWalletVerifier {
+            provider,
+            auth_data,
+        };
+
+        Ok(AuthRequestVerifier::new(verifier))
+    }
+
+    fn get_health_status(&self) -> eyre::Result<serde_json::Value> {
+        // Test the RPC connection to verify if the provider is healthy
+        // We're just checking if we can create a connection, but not actually making the request
+        let _client = JsonRpcClient::connect(&self.config.rpc_url);
```

1. `_client` isn't used at all here, we don't need the client?
2. if you had to reuse the client, why not store it in `NearWalletProvider`?

---

I would leave it out for now, if it fails let it fail when checking the account, as for the provider implementation, I would leave it as is so that we don't bloat it since only provider that needs this is NEAR if I remember correctly

---

yeah but what I meant was instead of calling

```rust
JsonRpcClient::connect(&self.config.rpc_url);
```

on every validation check, you can do that once on the near provider initialization and reuse the client, but this is a non-blocker so it's alright